src/compile/data/facet.ts

```diff
@@ -75,24 +64,47 @@ export class FacetNode extends DataFlowNode {
     return this.name;
   }
 
-  private assembleRowColumnData(channel: 'row' | 'column', crossedDataName: string): VgData {
-    const childChannel = channel === 'row' ? 'y' : 'x';
+  private getChildIndependentFieldWithStep() {
+    const childIndependentFieldsWithStep: ChildIndependentFieldsWithStep = {};
 
+    for (const channel of ['x', 'y'] as ScaleChannel[]) {
+      const childScaleComponent = this.childModel.component.scales[channel];
+      if (childScaleComponent && !childScaleComponent.merged) {
+        const type = childScaleComponent.get('type');
+        const range = childScaleComponent.get('range');
+
+        if (hasDiscreteDomain(type) && isVgRangeStep(range)) {
+          const domain = assembleDomain(this.childModel, channel);
```

Comment:
Yeah, you're not going to get the right domain here yet because we haven't finished assembling the data. We need to rename the data after we finish assembling the data.